packages/scripts/src/admin-proposals/change-final-fee/0_Propose.ts

```diff
@@ -0,0 +1,175 @@
+
+const hre = require("hardhat");
+
+import {
+  ERC20Ethers,
+  GovernorHubEthers,
+  GovernorRootTunnelEthers,
+  ParentMessengerBaseEthers,
+  ProposerV2Ethers,
+  StoreEthers,
+  VotingTokenEthers,
+} from "@uma/contracts-node";
+
+import { Provider } from "@ethersproject/abstract-provider";
+import { getGckmsSigner, getRetryProvider } from "@uma/common";
+import { BigNumberish, PopulatedTransaction, Signer, Wallet } from "ethers";
+import { BytesLike } from "ethers/lib/utils";
+import { getContractInstance, getContractInstanceByUrl, getContractInstanceWithProvider } from "../../utils/contracts";
+import { fundArbitrumParentMessengerForRelays, relayGovernanceMessages } from "../../utils/relay";
+import { tokensToUpdateFee } from "./common";
+
+interface AdminProposalTransaction {
+  to: string;
+  value: BigNumberish;
+  data: BytesLike;
+}
+
+const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+// Env vars
+const NODE_URL_ENV = "NODE_URL_";
+
+async function main() {
+  const adminProposalTransactions: AdminProposalTransaction[] = [];
+
+  let proposerSigner: Signer;
+
+  if (!process.env.NEW_FINAL_FEE_USD) throw new Error("NEW_FINAL_FEE_USD is not set");
+  if (!process.env.NEW_FINAL_FEE_WETH) throw new Error("NEW_FINAL_FEE_WETH is not set");
+  if (!process.env.UMIP_NUMBER) throw new Error("UMIP_NUMBER is not set");
+
+  const umipNumber = Number(process.env.UMIP_NUMBER);
+
+  const newFinalFeeUSD = Number(process.env.NEW_FINAL_FEE_USD);
+  const newFinalFeeWeth = Number(process.env.NEW_FINAL_FEE_WETH);
+
+  const arbitrumParentMessenger = await getContractInstance<ParentMessengerBaseEthers>("Arbitrum_ParentMessenger");
+
+  const governorRootTunnel = await getContractInstance<GovernorRootTunnelEthers>("GovernorRootTunnel"); // for polygon
+  const governorHub = await getContractInstance<GovernorHubEthers>("GovernorHub"); // rest of l2
+
+  const l2Networks = ["polygon", "arbitrum", "optimism"];
+  const l2NetworksNumber = { polygon: 137, optimism: 10, arbitrum: 42161 };
+
+  if (process.env.GCKMS_WALLET) {
+    proposerSigner = ((await getGckmsSigner()) as Wallet).connect(hre.ethers.provider as Provider);
+    if (proposerWallet.toLowerCase() != (await proposerSigner.getAddress()).toLowerCase())
+      throw new Error("GCKMS wallet does not match proposer wallet");
+  } else {
+    proposerSigner = (await hre.ethers.getSigner(proposerWallet)) as Signer;
+  }
+
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+
+  const proposer = await await getContractInstance<ProposerV2Ethers>("ProposerV2");
+  const store = await getContractInstance<StoreEthers>("Store");
+
+  // case mainnet
+  for (const token in tokensToUpdateFee) {
+    const provider = getRetryProvider(1) as Provider;
+
+    const isWeth = token == "WETH";
+    const newFinalFee = isWeth ? newFinalFeeWeth : newFinalFeeUSD;
+
+    const tokenAddress = tokensToUpdateFee[token as keyof typeof tokensToUpdateFee].mainnet;
```

Comment:
Nit: instead of accessing here, can you loop over Object.entries? Typescript may not handle this nicely -- if not no worries, just trying to avoid casts.